packages/flex-plugin-e2e-tests/src/utils/spawn.ts

```diff
@@ -0,0 +1,68 @@
+import { spawn } from 'child_process';
+
+import { logger } from 'flex-plugins-utils-logger';
+
+interface SpawnResult {
+  stdout: string;
+  stderr: string;
+}
+
+/**
+ * Promisified spawn
+ * @param cmd the command to spawn
+ * @param args the args to that command
+ */
+export default async (cmd: string, ...args: string[]): Promise<SpawnResult> => {
+  return new Promise((resolve, reject) => {
+    logger.info(`Running spawn command: **${cmd} ${args.join(' ')}**`);
+    const child = spawn(cmd, args);
+
+    const stdoutArr: Buffer[] = [];
+    const stderrArr: Buffer[] = [];
+
+    // errors
+    child.on('error', reject);
+    child.stdin.on('error', reject);
+    child.stdout.on('error', reject);
+    child.stderr.setEncoding('utf8');
+    child.stderr.on('error', reject);
+    child.stderr.on('data', (data) => {
+      if (typeof data === 'string') {
+        stderrArr.push(Buffer.from(data, 'utf-8'));
+      } else {
+        stderrArr.push(data);
+      }
+    });
+
+    // data
+    child.stdout.on('data', (data) => {
+      if (typeof data === 'string') {
+        stdoutArr.push(Buffer.from(data, 'utf-8'));
+      } else {
+        stdoutArr.push(data);
+      }
+    });
+
+    child.on('close', (code) => {
+      const stdout = Buffer.concat(stdoutArr).toString();
+      const stderr = Buffer.concat(stderrArr).toString();
+
+      if (code === 0) {
+        return resolve({ stdout, stderr });
+      }
+
+      return reject(new Error(`Command exited with code ${code} and message ${stdout}: ${stderr}`));
+    });
+  });
+};
+
+/**
+ * Helper for logging the result from a spawn
+ * @param result the result to log
+ */
+export const logResult = (result: SpawnResult): void => {
+  logger.info(result.stdout);
+  if (result.stderr) {
+    logger.warning(result.stderr);
```

Comment:
curious why the error is logged as a warning instead of error?

Reply:
`stderr` with code 0 means a warning, not an actual error.

Reply:
ahh

Reply:
where is the check for code 0? or is it just the default? 🤔

Reply:
In the `spawn` command https://github.com/twilio/flex-plugin-builder/pull/436/files#diff-fa31b28358da020711555aed155d512364708b3ffaea79575fddd9cdc9c5795aR50

Reply:
If this was an error, I reject the promise.